File: python/examples/hip3_order.py
#!/usr/bin/env python3
"""
HIP-3 Market Order Example

Trade on HIP-3 markets (community perps like Hypersea).
Same API as regular markets, just use "dex:symbol" format.

Requirements:
    pip install hyperliquid-sdk

Usage:
    export PRIVATE_KEY="0x..."
    python hip3_order.py
"""

import os
import sys

from hyperliquid_sdk import HyperliquidSDK

PRIVATE_KEY = os.environ.get("PRIVATE_KEY")

if not PRIVATE_KEY:
    print("Error: Set PRIVATE_KEY environment variable")
    print("  export PRIVATE_KEY='0xYourPrivateKey'")
    sys.exit(1)


def main():
    print("Hyperliquid HIP-3 Order Example")
    print("=" * 50)

    sdk = HyperliquidSDK(private_key=PRIVATE_KEY)
    print(f"Address: {sdk.address}")

    # List HIP-3 DEXes
    dexes = sdk.dexes()
    print("Available HIP-3 DEXes:")
    for dex in dexes[:5]:
        name = dex.get("name", dex) if isinstance(dex, dict) else str(dex)
        print(f"  {name}")

    # Trade on a HIP-3 market
    # Format: "dex:SYMBOL"
    # order = sdk.buy("xyz:SILVER", notional=11, tif="ioc")
    # print(f"HIP-3 order: {order}")

    print()
    print("HIP-3 markets use 'dex:SYMBOL' format")
    print("Example: sdk.buy('xyz:SILVER', notional=11)")


if __name__ == "__main__":
    main()

File: typescript/examples/hip3_order.ts
#!/usr/bin/env npx ts-node
// @ts-nocheck
/**
 * HIP-3 Market Order Example
 *
 * Trade on HIP-3 markets (community perps like Hypersea).
 * Same API as regular markets, just use "dex:symbol" format.
 */

import { HyperliquidSDK } from '@quicknode/hyperliquid-sdk';

const PRIVATE_KEY = process.env.PRIVATE_KEY;

if (!PRIVATE_KEY) {
  console.error("Set PRIVATE_KEY environment variable");
  process.exit(1);
}

async function main() {
  const sdk = new HyperliquidSDK(undefined, { privateKey: PRIVATE_KEY });

  // List HIP-3 DEXes
  const dexes = await sdk.dexes();
  console.log("Available HIP-3 DEXes:");
  for (const dex of dexes.slice(0, 5)) {
    console.log(`  ${dex.name || dex}`);
  }

  // Trade on a HIP-3 market
  // Format: "dex:SYMBOL"
  // const order = await sdk.buy("xyz:SILVER", { notional: 11, tif: "ioc" });
  // console.log(`HIP-3 order: ${order}`);

  console.log("\nHIP-3 markets use 'dex:SYMBOL' format");
  console.log("Example: sdk.buy('xyz:SILVER', { notional: 11 })");
}

main().catch(console.error);
